### PR TITLE
Modifications to sync with M.IM.WsTrust.

### PR DESCRIPTION
**Pull request by brentschmaltz**
WSFederationHttpBinding.CreateMessgeSecurity uses Security.Message.EstablishSecurityContext to control SCT bootstrap
Remove Target and EstablishSecurityContest properties from WSTrustTokenParameters
Use DiagnosticUtility to trace exceptions
Remove M.IM.Logging direct dependency
Add comments to Federation classes
Format comments using InvariantCulture

### CODE REVIEW

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs [hunk 1]
```diff
@@ -21,13 +18,13 @@ public class WSTrustChannelSecurityTokenManager : ClientCredentialsSecurityToken
         private WSTrustChannelClientCredentials _wsTrustChannelClientCredentials;
 
         /// <summary>
-        ///
+        /// Instaintiates a <see cref="WSTrustChannelSecurityTokenManager"/>.
         /// </summary>
```
Comment (mconnew) [inline src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs:22]: Typo on Instantiates

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelSecurityTokenProvider.cs [hunk 2]
```diff
@@ -27,7 +24,8 @@ namespace System.ServiceModel.Federation
 {
 
     /// <summary>
-    /// Custom WSTrustChannelSecurityTokenProvider that returns a SAML assertion
+    /// <see cref="WSTrustChannelSecurityTokenProvider"/> has been designed to work with <see cref="WSFederationHttpBinding"/> to that will send a WsTrust message to obtain a token from an STS and add the token as
+    /// an issued token when communicating with a WCF relying party.
```
Comment (mconnew) [inline src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelSecurityTokenProvider.cs:28]: I think you merged two different phrases in your head. "to that will send" doesn't make sense.

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustTokenParameters.cs [hunk 3]
```diff
@@ -24,20 +25,25 @@ public class WSTrustTokenParameters : IssuedSecurityTokenParameters
         private TimeSpan _maxIssuedTokenCachingTime = DefaultMaxIssuedTokenCachingTime;
         private MessageSecurityVersion _messageSecurityVersion;
         private int _issuedTokenRenewalThresholdPercentage = DefaultIssuedTokenRenewalThresholdPercentage;
-        private string _target;
 
         /// <summary>
-        /// Values that are used to obtain a token from an IdentityProvider
+        /// Instantiates aa <see cref="wsTrustTokenParameters"/> that describe the parameters for a WSTrust request.
         /// </summary>
```
Comment (mconnew) [inline src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustTokenParameters.cs:31]: typo: aa

File: src/System.Private.ServiceModel/tests/Scenarios/Binding/WS/FederationHttp/WSFederationHttpBindingTests.cs [hunk 4]
```diff
@@ -32,7 +32,6 @@ public static void WSFederationHttpBindingTests_Succeeds()
         {
             // *** SETUP *** \\
             issuerAddress = new EndpointAddress(new Uri(Endpoints.WSFederationAuthorityLocalSTS));
-            tokenTargetAddress = Endpoints.Https_SecModeTransWithMessCred_ClientCredTypeIssuedTokenSaml2;
             serviceEndpointAddress = new EndpointAddress(new Uri(tokenTargetAddress));
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Scenarios/Binding/WS/FederationHttp/WSFederationHttpBindingTests.cs:35]: This broke the test as it's used on the next line to create the serviceEndpointAddress.

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs:22]: fixed in: b160df7f33caf6447f14e47b81ab3ff9f5930924

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelSecurityTokenProvider.cs (hunk 2 above)
Comment (brentschmaltz) [reply src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelSecurityTokenProvider.cs]: fixed in: b160df7f33caf6447f14e47b81ab3ff9f5930924

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustTokenParameters.cs (hunk 3 above)
Comment (brentschmaltz) [reply src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustTokenParameters.cs:31]: fixed in: b160df7f33caf6447f14e47b81ab3ff9f5930924

File: src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustChannelSecurityTokenManager.cs:22]: fixed in: 9a8f277d85bf9419167870cad953b59d0d7a6559

File: src/System.Private.ServiceModel/tests/Scenarios/Binding/WS/FederationHttp/WSFederationHttpBindingTests.cs (hunk 4 above)
Comment (brentschmaltz) [reply src/System.Private.ServiceModel/tests/Scenarios/Binding/WS/FederationHttp/WSFederationHttpBindingTests.cs]: fixed in: 9a8f277d85bf9419167870cad953b59d0d7a6559

**Comment by mconnew** [comment]
I don't know how this build on the PR machine. I cloned your repo locally and I'm getting lots of errors on the restore step.  
```
error NU1103: Unable to find a stable package Microsoft.IdentityModel.Xml with version (>= 6.8.0) [C:\git\wcf-brent\System.ServiceModel.sln]
error NU1103:   - Found 18 version(s) in myget [ Nearest version: 6.8.1-preview-11014204844 ] [C:\git\wcf-brent\System.ServiceModel.sln]
error NU1103:   - Found 0 version(s) in dotnet-eng [C:\git\wcf-brent\System.ServiceModel.sln]error NU1103:   - Found 0 version(s) in dotnet-tools [C:\git\wcf-brent\System.ServiceModel.sln]
error NU1103: Unable to find a stable package Microsoft.IdentityModel.Tokens.Saml with version (>= 6.8.0) [C:\git\wcf-brent\System.ServiceModel.sln]
error NU1103:   - Found 18 version(s) in myget [ Nearest version: 6.8.1-preview-11014204844 ] [C:\git\wcf-brent\System.ServiceModel.sln]
error NU1103:   - Found 0 version(s) in dotnet-eng [C:\git\wcf-brent\System.ServiceModel.sln]error NU1103:   - Found 0 version(s) in dotnet-tools [C:\git\wcf-brent\System.ServiceModel.sln]
```

**Comment by mconnew** [comment]
I found the source of the errors. Apparently our build system removes nuget.org as a package source when the tests are targeting .net5. This only affects local building as the package restore is done differently for our PR and CI system.

**Comment by mconnew** [comment]
Closing in preference of #4400 which contains all these commits.

**Comment by brentschmaltz** [comment]
@mconnew thanks.

**Pull request closed without merging**